src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx

```diff
@@ -0,0 +1,497 @@
+    super(props, context);
+
+    this.state = {
+      formErrors: [],
+      formErrorsByField: { ...defaultValidation },
+      dataSourceTitle: '',
+      dataSourceDescription: '',
+      endpoint: '',
+      showCreateCredentialModal: false,
+      selectedCredentialSourceType: credentialSourceOptions[1].value,
+      selectedCredentials: [],
+      availableCredentials: [],
+      createCredential: defaultCreateCredentialsForm,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchAvailableCredentials();
+  }
+
+  fetchAvailableCredentials() {
+    const { savedObjects } = this.context.services;
+    this.props.displayLoadingMask(true);
+    getExistingCredentials(savedObjects.client)
+      .then((fetchedCredentials: CredentialsComboBoxItem[]) => {
+        if (fetchedCredentials?.length) {
+          this.setState({ availableCredentials: fetchedCredentials });
+        }
+      })
+      .catch(() => {
+        this.props.displayToastMessage({
+          id: 'dataSourcesManagement.createDataSource.fetchExistingCredentialsFailMsg',
+          defaultMessage: 'Error while finding existing credentials.',
+          color: 'warning',
+          iconType: 'alert',
+        });
+      })
+      .finally(() => {
+        this.props.displayLoadingMask(false);
+      });
+  }
+
+  /* Validations */
+
+  isFormValid = () => {
+    const { formErrors, formErrorsByField } = performDataSourceFormValidation(this.state);
+
+    this.setState({
+      formErrors,
+      formErrorsByField,
+    });
+
+    return formErrors.length === 0;
+  };
+
+  /* Events */
+
+  onChangeTitle = (e: { target: { value: any } }) => {
+    this.setState({ dataSourceTitle: e.target.value }, () => {
+      if (this.state.formErrorsByField.title.length) {
+        this.isFormValid();
+      }
+    });
+  };
+
+  onChangeDescription = (e: { target: { value: any } }) => {
+    this.setState({ dataSourceDescription: e.target.value }, () => {
+      if (this.state.formErrorsByField.description.length) {
+        this.isFormValid();
+      }
+    });
+  };
+
+  onChangeEndpoint = (e: { target: { value: any } }) => {
+    this.setState({ endpoint: e.target.value }, () => {
+      if (this.state.formErrorsByField.endpoint.length) {
+        this.isFormValid();
+      }
+    });
+  };
+
+  onChangeCredentialSourceType = (value: string) => {
+    /* reset already selected existing credential */
+    if (
+      this.state.selectedCredentials.length &&
+      value !== CredentialSourceType.ExistingCredential
+    ) {
+      this.resetExistingCredentialSelection();
+    }
+
+    this.setState({ selectedCredentialSourceType: value }, () => {
+      if (this.state.formErrors.length) {
+        this.isFormValid();
+      }
+    });
+  };
+
+  resetExistingCredentialSelection = () => {
+    this.setState({
+      selectedCredentials: [],
+      availableCredentials: this.state.availableCredentials?.map((rec) => {
```

Comment:
nit: rec => record? if that's what it means :)

Reply:
Done. Refactored